src/main/java/io/neonbee/internal/scanner/ClassPathScanner.java

```diff
@@ -91,160 +96,197 @@ public static ClassLoader getClassLoader() {
      *
      * This allows to e.g. denote certain classes or resources of the JAR files to be exposed.
      *
+     * @param vertx         the Vert.x instance
      * @param attributeName The name of the attribute in the MANIFEST.MF to search for.
-     * @return a list of strings (separated list of manifest attribute values)
-     * @throws IOException If an I/O error occurs while scanning the class paths
+     * @return a future of a list of strings (separated list of manifest attribute values)
      */
-    public List<String> scanManifestFiles(String attributeName) throws IOException {
-        List<String> resources = new ArrayList<>();
-        for (URL manifestResources : getManifestResourceURLs()) {
-            try (InputStream inputStream = manifestResources.openStream()) {
-                Manifest manifest = new Manifest(inputStream);
-                // Attribute looks like: Attribute-Name: package.Resource1; package.Resource2
-                String attributeValue = manifest.getMainAttributes().getValue(attributeName);
-                if (!Strings.isNullOrEmpty(attributeValue)) {
-                    SEPARATOR_PATTERN.splitAsStream(attributeValue).map(String::trim).forEach(resources::add);
+    public Future<List<String>> scanManifestFiles(Vertx vertx, String attributeName) {
+        return vertx.executeBlocking(promise -> {
+            try {
+                List<String> resources = new ArrayList<>();
+                for (URL manifestResources : getManifestResourceURLs()) {
+                    try (InputStream inputStream = manifestResources.openStream()) {
+                        Manifest manifest = new Manifest(inputStream);
+                        // Attribute looks like: Attribute-Name: package.Resource1; package.Resource2
+                        String attributeValue = manifest.getMainAttributes().getValue(attributeName);
+                        if (!Strings.isNullOrEmpty(attributeValue)) {
+                            SEPARATOR_PATTERN.splitAsStream(attributeValue).map(String::trim).forEach(resources::add);
+                        }
+                    }
                 }
+                promise.complete(resources);
+            } catch (IOException e) {
+                promise.fail(e);
             }
-        }
-        return resources;
+        });
     }
 
     /**
      * Scans the whole class path for annotations on the whole class, shorthand for: <br>
      * {@code scanForAnnotation(Class, ElementType.TYPE)}.
      *
-     * @see #scanForAnnotation(Class, ElementType...)
+     * @see #scanForAnnotation(Vertx, Class, ElementType...)
+     * @param vertx           the Vert.x instance
      * @param annotationClass the annotation to check for
-     * @return a List of resources on the class path
-     * @throws IOException        If operations on the filesystem fail
-     * @throws URISyntaxException If parsing the URI fails
+     * @return a future to a list of resources on the class path
```

Comment:
because above we also habe "of" a list, but below in all the methods you have "to". Maybe a copy paste thing, but if not we should also change it above.

Reply:
I think it is always "a future to", because it "points" in the future.